spec/lib/data_update_scripts/backfill_community_emoji_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "rails_helper"
+require Rails.root.join(
+  "lib/data_update_scripts/20220201202226_backfill_community_emoji.rb",
+)
+
+describe DataUpdateScripts::BackfillCommunityEmoji do
+  it "does not update the community_name if the community does not have a community_emoji" do
+    allow(Settings::Community).to receive(:community_name).and_return("Emoji less Community")
+    allow(Settings::Community).to receive(:community_emoji).and_return(nil)
+
+    expect do
+      described_class.new.run
+    end.not_to change(Settings::Community, :community_name)
+  end
+
+  it "updates the community_name if the community has a community_emoji" do
+    Settings::Community.community_name = "Emoji Community"
+    Settings::Community.community_emoji = "🥐"
```

Comment:
I tried doing this:
```
    allow(Settings::Community).to receive(:community_name).and_return("Emoji Community")
    allow(Settings::Community).to receive(:community_emoji).and_return("🥐")
```

but the expectation fails when stubbing it out in this way. Whilst I think the approach taken above is okay (but maybe not optimal) I've seen it set like this in other DUS files, I'm curious to know if anyone knows why its failing?

Reply:
It seems to me like the `allow` stub is used to return the original value `"Emoji Community"` string even after the DUS executes. So this happens:

1. Both `community_name` and `community_emoji` are stubbed with `allow` to return the original values
2. The DUS runs, and assigns the new value
3. The `expect` assertion fails because `Settings::Community.community_name` will continue to return the stubbed value despite the DUS ran

I looked up options and there's a block syntax for the `allow` stubbing but it doesn't work if we used `and_return(...)`, so I'm not sure if there's a way to achieve this. Maybe we just have to deal with assigning them manually in these specs. I actually think it would be best if we do that for all these specs (not just this one where the name should change), to ensure the `expect` clauses are testing the actual result instead of the original stubbed value.

Reply:
Cool thanks Fernando, I'm going to leave it as is for now then if nobody else has any other suggestions.